lex.go

```diff
@@ -370,32 +370,33 @@ func lexExpr(l *lexer) stateFn {
 
 // lexName is the state function for lexing names.
 func lexName(l *lexer) stateFn {
-	var s string
+	var typ itemType
 
-	typ := itemUnquotedLiteral
+	switch l.next() {
+	case '$':
+		typ = itemVariable
+	case '.':
+		typ = itemKeyword
+	default:
+		typ = itemUnquotedLiteral
 
-	for {
-		r := l.next()
+		l.backup() // backup to the first rune
+	}
 
-		switch {
-		default:
-			l.backup()
+	var (
+		s string // item value
+		r rune   // current rune
+	)
```

Comment:
```suggestion
	var s string // item value
```

Reply:
cannot do that, r is needed outside the loop